### ISSUE TITLE
Not working with Rekordbox 7.0.1

### ISSUE THREAD

**Issue by AlexEneas** [Unreal-Dan/RekordBoxSongExporter#27]
I'm sure you're already aware, but this isn't working with Rekordbox 7

**Comment by Unreal-Dan** [comment]
Thanks for letting me know, I don't actually use it anymore so anytime you guys need updates you'll have to let me know like this.

I will download 7 and give it a look over this weekend and deploy and update, thanks again

**Comment by Unreal-Dan** [comment]
A small update, I am working on this but a lot has changed and I'm having trouble locating one of the rekordbox functions that I originally hooked. I might need to find another way to implement the same hook with a different function for 7+

**Comment by getaped1** [comment]
Hello,
Have you further tried to get this to work in rb7?

**Comment by Unreal-Dan** [comment]
just a small update, life gets in the way, I will look into this again and I haven't abandoned the project.

**Comment by Zethrus** [comment]
Any updates for this?

**Comment by Unreal-Dan** [comment]
I've been quite busy lately, how badly do you guys need it updated for rb7? 

I tried to take a look at it back when rb7 was released but it seems like some of the functions I used for my hooks have changed and so I would need to do some reverse engineering work to figure out a fix.

I will try to find some time, between work and my current projects it's hard to get back into this. 

To be honest I made this for an ex, she did me dirty and burned the bridge so I don't have much motivation to work on it anymore, the only reason I came back and updated it is because so many people found it useful.

**Comment by Zethrus** [comment]
> I've been quite busy lately, how badly do you guys need it updated for rb7?
> 
> I tried to take a look at it back when rb7 was released but it seems like some of the functions I used for my hooks have changed and so I would need to do some reverse engineering work to figure out a fix.
> 
> I will try to find some time, between work and my current projects it's hard to get back into this.
> 
> To be honest I made this for an ex, she did me dirty and burned the bridge so I don't have much motivation to work on it anymore, the only reason I came back and updated it is because so many people found it useful.

Well, rekordbox 7 is quite widely used now over rekordbox 6 I think. It's on version 7.0.8 last I checked, an application like this is essential for me to grab data for OBS text overlays so, yeah it would be amazing if it could be updated.

There is other software out there that does similar thing such as "unbox" but even that isn't working currently with rekordbox 7, at least that I can figure out.
GitHub issues on that project "unbox" pointed me over to this project here of yours, which in hopes thought it would work for rekordbox 7 with no luck :(

**Comment by Unreal-Dan** [comment]
Hahaha I see, unbox was using my module for instant updates.

Originally unbox scraped the rbox database so it had a delay and potential for database corruption, I let him use my module for instant updates but I assumed he always had a fallback if mine didn't work. 

Let me look into it this weekend and see how far I can get

**Comment by Zethrus** [comment]
Yeah, something like that it seems. I guess that would be why unbox also hasn't had an update for quite a while then. 

I appreciate you looking into this!

**Comment by Unreal-Dan** [comment]
I spent some time and managed to make quite a bit of progress, the progress is in this branch/PR here in case anybody wants to use it right away. It is able to log things like BPM but track information is broken, I'll fix that next and it should be good to go.

https://github.com/Unreal-Dan/RekordBoxSongExporter/pulls 

Spent many hours sorting this out so I'm done for today, check back in tomorrow for more updates

**Comment by Unreal-Dan** [comment]
[RekordBoxSongExporter4.0-early-snapshot.zip](https://github.com/user-attachments/files/18391022/RekordBoxSongExporter4.0-early-snapshot.zip)

Actually I figured it out rather quickly, here is a snapshot of the version working for rbox 7.0.8, please give it some testing and let me know if there's any issues. I think it's working good but I may have missed some things.

**Comment by Unreal-Dan** [comment]
I spent a solid all-nighter and managed to make a lot of progress, 


Here is a snapshot of the latest version working good with rbox 7.0.8, please give it some testing and let me know if there's any issues.

@AlexEneas @Zethrus @getaped1

**Comment by Zethrus** [comment]
> I spent a solid all-nighter and managed to make a lot of progress,
> 
> 
> Here is a snapshot of the latest version working good with rbox 7.0.8, please give it some testing and let me know if there's any issues.
> 
> @AlexEneas @Zethrus @getaped1

So far, for what I am using it for it seems to be working great.

Thank you so much for the quick update! I will let you know if I find any bugs at a later date during Twitch streams.

**Comment by getaped1** [comment]
> I spent a solid all-nighter and managed to make a lot of progress,
> 
> 
> Here is a snapshot of the latest version working good with rbox 7.0.8, please give it some testing and let me know if there's any issues.
> 
> @AlexEneas @Zethrus @getaped1

Ok, I spent about 4 hours testing tonight. I used these placeholders: artist, title, bpm, master_bpm, key, time, rt_master_bpm

master_bpm & rt_master_bpm were the only 2 that did not work. Both give 0.

Thanks for the update!

**Comment by Unreal-Dan** [comment]
[RekordBoxSongExporter4.0-early-snapshot-2.zip](https://github.com/user-attachments/files/18465137/RekordBoxSongExporter4.0-early-snapshot-2.zip)

Give this a go, it should fix the bpm and a few other things. 

How do you guys feel about the module throwing a popup on injection giving myself credit?  I don't want to be annoying but I also think it's fair people know unbox uses my module and where to find it. 

I didn't implement it in that early snapshot but I'm heavily considering it for the next full release, if I do I'll delete those early snapshot uploads and ask that you helpful fellows keep it for personal use.

**Comment by Unreal-Dan** [comment]
released latest, removing old snapshots, thanks guys.
